mongo/oplog.go

```diff
@@ -236,39 +225,38 @@ func (t *OplogTailer) loop() error {
 	// See: http://docs.mongodb.org/v2.4/reference/bson-types/#timestamps
 	var idsForLastTimestamp []int64
 
+	newIter := func() Iterator {
+		return t.session.NewIter(lastTimestamp, idsForLastTimestamp)
+	}
+	iter := newIter()
+	defer func() { iter.Close() }() // iter may be replaced, hence closure
```

Comment:
given the later check about iter == nil shouldn't we check that here so we don't try to close a nil pointer?

Reply:
I guess the idea is to never have a nil iterator. does it cause an unnecessary query to the db if we're stopping/have nothing to do?

Reply:
Right, it'll always be non-nil (the `iter == nil` check was removed).

It's pretty unlikely that the tailer would be killed before it gets to this point, but even if it did, the cost will be negligible.